taar/recommenders/locale_recommender.py

```diff
@@ -25,11 +25,14 @@ def __init__(self):
         if self.top_addons_per_locale is None:
             logger.error("Cannot download the top per locale file {}".format(ADDON_LIST_KEY))
 
-    def can_recommend(self, client_data):
+    def can_recommend(self, client_data, extra_data={}):
         # We can't recommend if we don't have our data files.
         if self.top_addons_per_locale is None:
             return False
-        client_locale = client_data.get('locale', None)
+
+        # If we have data coming from other sources, we can use that for
+        # recommending.
+        client_locale = client_data.get('locale', None) or extra_data.get('locale', None)
```

Comment:
We must also check if the value contained in `extra_data.get('locale', None)` is found in `self.top_addons_per_locale`

Reply:
That's done below, in the checks at lines 39-43